CHANGELOG.md

```diff
@@ -9,6 +9,10 @@
 - Added Travis CI config for pylint and pyflakes.
 - Added unittest scaffolding with nose
 - Added `AssignerTestCase`, tests for `assigner`, `assigner get`, and all `--help` uses.
+- Added BackendBase, RepoBase, StudentRepoBase, and TemplateRepoBase.
+- Separated Gitlab backend from most of the code. Added Base implementations for Gitlab.
+- Added require_backend decorator with backend config option to load the desired backend.
+- pyflakes 1.6.0 upgrade for type annotations fixes.
```

Comment:
Let's add a note that "Base Repos" are now known as "Template Repos".

Reply:
Done.